src/microbe_stage/MicrobeStage.cs

```diff
@@ -866,22 +866,44 @@ private void OnPlayerNoticeMessage(Microbe player, IHUDMessage message)
             GD.PrintErr("Chemoreception data reported for non-player cell");
 
         int currentLineIndex = 0;
-        var position = microbe.GlobalTransform.origin;
+        var colonyMicrobes = microbe.Colony?.ColonyMembers ?? new List<Microbe> { microbe };
```

Comment:
This is not really what I had in mind for this method. I think a simpler way to implement this, which I think is good enough and would give better performance and less confusing overlapping lines, is that just the player cell would return the detections it wants. So reverting a bunch of the changes here and then modifying `Microbe.GetDetectedCompounds` to take into account all the non-duplicate detections from the colony would be ideal in my opinion. That has the slight drawback that the lines wouldn't becoming out of the cells that have the detectors, but I think that is actually a benefit if the player has a chemoreceptor on each cell, that will end up with a huge mess of lines going everywhere.